src/Indexer.js

```diff
@@ -79,7 +93,10 @@ export default class Indexer {
         const indexExists = await this.client.indices.exists({ index: index })
 
         if (!indexExists) {
-          await this.client.indices.create({ index: index })
+          // analysis and filter settings must be provided at index creation time; alternatively, the index can be closed, configured, and reopened.
+          // otherwise, an error is thrown along the lines of "error setting up indices: [illegal_argument_exception] Can't update non dynamic settings"
+          // https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/6.x/api-reference.html#_indices_create
+          await this.client.indices.create({ index: index, body: { settings: this.indexSettings() } })
```

Comment:
the docs indicate that the `mappings` (in addition to the `settings`) can be configured in this `create` call.  i wasn't sure whether to refactor that out of the code below, or to go for the least disruptive thing.  since it wasn't clear to me whether the `type` info that's provided below could be provided in the `create` call above, i went with the less disruptive approach.  if it is possible to fold exactly what's being done below into the `create` call above i'd be in favor of doing so (fewer network calls/less opportunity for failure, and makes just as much sense semantically, while being slightly more concise/readable IMHO).